commands/deploy_test.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Alex Ellis 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"testing"
+
+	"net/http"
+
+	"github.com/openfaas/faas-cli/test"
+)
+
+func Test_deploy(t *testing.T) {
+	s := test.MockHttpServer(t, []test.Request{
+		{
+			Method:             http.MethodDelete,
+			Uri:                "/system/functions",
+			ResponseStatusCode: http.StatusOK,
+		},
+		{
+			Method:             http.MethodPost,
+			Uri:                "/system/functions",
+			ResponseStatusCode: http.StatusOK,
+		},
+	})
+	defer s.Close()
+
+	faasCmd.SetArgs([]string{
+		"deploy",
+		"--gateway=" + s.URL,
+		"--image=golang",
+		"--name=test-function",
+	})
+	faasCmd.Execute()
```

Comment:
How do you verify the result in this test?

Reply:
With something like this:

`
test.CaptureStdout(func(){})

r := regexp.MustCompile(`(?m:function-test-[12])`)
	matches := r.FindAllStringSubmatch(stdOut, 2)
	if len(matches) != 2 {
		t.Fatalf("Output is not as expected:\n%s", stdOut)
	}
`

Reply:
@alexellis Just updated this test with output verification